### PR TITLE
Watermelon using app token (team tool)

### PR DESCRIPTION
**Pull request by EstebanDalelR**
## Description
Uses the app token instead of the user token to get info
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)  
- [x] New feature (non-breaking change which adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)  
- [ ] Documentation update  
- [ ] Chore: cleanup/renaming, etc  
- [ ] RFC  
## Acceptance

- [x] I have read [the Contributing guidelines](CONTRIBUTING.md)
- [x] I have read the [Code of Conduct](CODE_OF_CONDUCT.md)

### CODE REVIEW

File: utils/actions/getAllServices.ts [hunk 1]
```diff
@@ -71,7 +75,7 @@ export default async function getAllServices({
     });
     return { error: error.message };
   }
-  const [github, jira, confluence, slack, notion, linear, asana] =
+  const [github, teamGitHub, jira, confluence, slack, notion, linear, asana] =
```
Comment (baristaGeek) [inline utils/actions/getAllServices.ts:78]: Nice idiomatic var name

**Review by baristaGeek — COMMENTED** [review]
Testing notes:

- Tested with the previously successful payload on #324 and it works 
- Tested on #328 after deleting estebanvargas94 (username associated to the estebanvargasgarcia1994@gmail.com email) with the previously successful payload that I used to create the PR. Got a new comment asking me to login 
Screen Shot 2023-10-13 at 12 22 43 PM

File: app/api/actions/github/route.ts [hunk 2]
```diff
@@ -311,7 +314,7 @@ export async function POST(request: Request) {
                 owner,
                 issue_number: number,
                 repo,
-                body: "[Please login to Watermelon to see the results](https://app.watermelontools.com/)",
+                body: "[Please login to GitHub in Watermelon to see the results](https://app.watermelontools.com/)",
```
Comment (EstebanDalelR) [inline app/api/actions/github/route.ts:317]: Fix to the text, there is no way to find a user without them logging in.
@baristaGeek

**Review by baristaGeek — APPROVED** [review]
Approving understanding the technical restrictions that we have.

**Review by baristaGeek — CHANGES_REQUESTED** [review]
We must handle this case. I used my estebanvargasgarcia1994@gmail.com account. 

Screen Shot 2023-10-24 at 12 09 55 PM

**Review by baristaGeek — CHANGES_REQUESTED** [review]
Ran with baristaGeek (estebanvargas94@gmail.com) which is part of the team and the team's query count didn't increase

https://github.com/watermelontools/watermelon/assets/8325094/bd93818a-b267-4f8f-bf42-c6b33288988e

**Review by baristaGeek — CHANGES_REQUESTED** [review]
This user flow should behave as follows:
1. Opening a PR with estebanvargasgarcia1994@gmail.com (a user not part of the dbo.github table but part of the team)
2. On the remote branch (feature/team-GH-fetch)
3. PR is contextualized and pre-reviewed

https://github.com/watermelontools/watermelon/assets/8325094/226aac6a-6ba4-4cfc-88a2-c3677e15db7e

**Comment by baristaGeek** [comment]
> We must handle this case. I used my [estebanvargasgarcia1994@gmail.com](mailto:estebanvargasgarcia1994@gmail.com) account.
> 
> Screen Shot 2023-10-24 at 12 09 55 PM

This is working properly now

**Comment by baristaGeek** [comment]
> Ran with baristaGeek ([estebanvargas94@gmail.com](mailto:estebanvargas94@gmail.com)) which is part of the team and the team's query count didn't increase
> 
>  Screen.Recording.2023-10-24.at.12.51.45.PM.mov

If this improves cool, but it's not a priority

**Comment by baristaGeek** [comment]
I think we're very close! I didn't get "user not registered" as the last time and actually got a successful response. 

Is the only thing left to do actually executing the commenting -> detecting console logs -> labeling the PR flow? 

Screen Shot 2023-10-25 at 5 23 54 PM

**Review by baristaGeek — APPROVED** [review]
Approving because it's working better than the current version of #334 (I recognize there are merge conflicts). The hard part works :tada: :tada:

We just have to fix `placeholder={`Teammate@${teamName.toLowerCase()}.com`}` which is something minor

**Comment by EstebanDalelR** [comment]
> We just have to fix `placeholder={`Teammate@${teamName.toLowerCase()}.com`}` which is something minor

Weird, this line hasn't ever been touched.

**Comment by EstebanDalelR** [comment]
Anyway I opened #342 with that.

**Comment by baristaGeek** [comment]
> > We just have to fix `placeholder={`Teammate@${teamName.toLowerCase()}.com`}` which is something minor
> 
> Weird, this line hasn't ever been touched.

You're right this is broken in prod. Let's fix it. For now let's merge this

**Pull request merged**